src/appleseed.python/studio/helpers.py

```diff
@@ -0,0 +1,35 @@
+import appleseed as asr
+import appleseed.studio as studio
+
+def get_textures(container):
+    assert isinstance(container, asr.BaseGroup)
+
+    textures = list(container.textures())
+
+    assemblies = container.assemblies()
+    for key in assemblies:
+        textures += get_textures(assemblies[key])
+
+    return textures
+
+
+def convert_all_textures_to_tx(maketx_path):
+    scene = studio.current_project().get_scene()
+    textures = get_textures(scene)
+
+    tx_converter = asr.TXConverter(maketx_path)
+
+    for texture in textures:
+        texture_parameters = texture.get_parameters()
+        texture_path = texture_parameters['filename']
```

Comment:
You need to only collect textures of types you know how to deal with, i.e. `texture2d`, because not all texture types with have a `filename` parameter.